app/src/main/java/ch/epfl/sdp/gameOver/GameOverActivity.java

```diff
@@ -2,28 +2,41 @@
 
 import android.content.Intent;
 import android.os.Bundle;
+import android.widget.TextView;
 
 import androidx.appcompat.app.AppCompatActivity;
 
-import ch.epfl.sdp.MainActivity;
+import ch.epfl.sdp.MainMenuActivity;
 import ch.epfl.sdp.R;
+import ch.epfl.sdp.dependencies.AppContainer;
+import ch.epfl.sdp.dependencies.MyApplication;
+import ch.epfl.sdp.entity.PlayerManager;
 import ch.epfl.sdp.utils.JunkCleaner;
 
 /**
  * @brief displays splash-screen-like game over screen that lasts for a few seconds
  */
 public class GameOverActivity extends AppCompatActivity {
+    private TextView gameOverText;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_game_over);
-        findViewById(R.id.backFromGameOverButton).setOnClickListener(v -> goToMainMenu());
-        JunkCleaner.clearAll();
+        gameOverText = findViewById(R.id.gameOverText);
+        if (PlayerManager.getInstance().getCurrentUser().getHealthPoints() > 0) {
+            gameOverText.setText(R.string.winnerText);
+        }
+        findViewById(R.id.backFromGameOver).setOnClickListener(v -> goToMainMenu());
```

Comment:
in findViewById u used v-> and in some this:: does it matter? If its equal than we have to be consitent!

Reply:
Sadly, we cannot here, because goToMainMenu does not take a view as argument! (we could give an unused argument, but that would not be useful)